client/lib/tree-select/index.js

```diff
@@ -0,0 +1,72 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { isObject, some, isFunction } from 'lodash';
+
+/**
+ * Returns a selector that caches values.
+ *
+ * @param  {Function} getDependents A Function describing the dependent(s) of the selector.
+ *                                    Must return an array which gets passed as the first arg to the selector
+ * @param  {Function} selector      A standard selector for calculating cached result
+ * @return {Function}               Cached selector
+ */
+export default function treeSelect( getDependents, selector ) {
+	if ( ! isFunction( getDependents ) || ! isFunction( selector ) ) {
```

Comment:
This check could be done only in development mode.

Reply:
ok, adding check

Reply:
Also added a unit test for this in https://github.com/Automattic/wp-calypso/pull/20547/files#diff-e1fde4b19690dc0c18ea9cd84f272397R90